### PR TITLE
Improve UI handling of paymaster transactions

### PR DESCRIPTION
**Pull request by everdimension**
Handle loading state when we fetch nonce;
Display zero fee in History Action Item;

### CODE REVIEW

File: src/ui/components/SignTransactionButton/SignTransactionButton.tsx [hunk 1]
```diff
@@ -68,13 +70,16 @@ export const SignTransactionButton = React.forwardRef(
...
     useImperativeHandle(ref, () => ({ sendTransaction }));
 
+    const isLoading = isLoadingProp || sendTxMutation.isLoading;
+    const isSending = sendTxMutation.isLoading;
+
     return isDeviceAccount(wallet) ? (
       <HardwareSignTransaction
         ref={hardwareSignRef}
         derivationPath={wallet.derivationPath}
-        isSending={sendTxMutation.isLoading}
```
Comment (zerts) [inline src/ui/components/SignTransactionButton/SignTransactionButton.tsx:75]: This should be merged accurately in hold to sign button

File: src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx [hunk 2]
```diff
@@ -41,24 +41,36 @@ export function FeeLine({
...
-                symbol: nativeAsset.symbol,
-              }}
-              address={address}
-            />
-          ) : (
-            nativeAsset.symbol?.toUpperCase()
-          )}
-          <span>({formatCurrencyValue(feeCurrency, 'en', 'usd')})</span>
-        </HStack>
+        {fee.quantity == '0' ? (
```
Comment (zerts) [inline src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx:46]: is ` == '0'` a typo or is it something like `== null`?

File: src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx [hunk 3]
```diff
@@ -41,24 +41,36 @@ export function FeeLine({
...
+                  asset_code: nativeAsset.id,
+                  name: nativeAsset.name,
+                  symbol: nativeAsset.symbol,
+                }}
+                address={address}
+              />
+            ) : (
+              nativeAsset.symbol?.toUpperCase()
+            )}
+            <span>({formatCurrencyValue(feeCurrency, 'en', 'usd')})</span>
```
Comment (zerts) [inline src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx:71]: probably, this needs to be merged accurately with currency settings as well

File: src/ui/components/SignTransactionButton/SignTransactionButton.tsx [hunk 4]
```diff
@@ -85,15 +90,12 @@ export const SignTransactionButton = React.forwardRef(
         onClick={onClick}
         render={({ handleClick }) => (
           <Button
-            disabled={sendTxMutation.isLoading}
+            disabled={isLoading}
```
Comment (zerts) [inline src/ui/components/SignTransactionButton/SignTransactionButton.tsx:93]: Can there be a problem with disabled option being overwritten by `buttonProps.disabled = false` ?

Comment (everdimension) [reply src/ui/components/SignTransactionButton/SignTransactionButton.tsx:93]: Hm, good catch, I'll add comment

File: src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx (hunk 2 above)
Comment (everdimension) [reply src/ui/pages/History/ActionDetailedView/components/FeeLine.tsx:46]: Intentional, but fortunately backend introduced a dedicated property for this, so I'll update

**Review by zerts — APPROVED** [review]

**Pull request merged**